<!-- launches joint trajectory action server using the kinova drivers -->
<launch>

    # TODO the default should be optional as soon as first round of testing is finished 
    <arg name="load_ros_params" default="true"/>
    <arg name="names_config_file" value="$(find jaco_joints)/config/JacoJoints.yaml"/>

    <include if="$(arg load_ros_params)" file="$(find jaco_joints)/launch/init_jaco_joints.launch">
        <arg name="names_config_file" value="$(arg names_config_file)"/>
    </include>
	
	# velocity limitation (in RAD per second) for joints 1,2,3 and 4,5,6	
	# This only applies if use_angle_poses is set to true, and then indicates the speed at which the trajectory is played.
	<arg name="max_speed_123" default="0.3"/>
	<arg name="max_speed_456" default="0.3"/>
	
	# set to true to use angle poses in a trajectory to control the device. If this is false,
	# velocities are used instead.
	<arg name="use_angle_poses" default="true"/>
	
	<!-- JointTrajectoryAction action topic name -->
	<arg name="joint_trajectory_action_topic" default="jaco/joint_trajectory_action" />
	
	<!-- joint_states topic, this is where sensor_msgs::JointState messages of the whole arm including fingers are published -->
    <arg name="joint_states_topic" default="jaco/joint_states" />

 
	<node name="jaco_trajectory_action_kinova" pkg="jaco_kinova" type="jaco_trajectory_action_server_kinova_node" respawn="false" output="screen">
		<param name="joint_trajectory_action_topic" value="$(arg joint_trajectory_action_topic)" />
		<param name="joint_states_topic" value="$(arg joint_states_topic)" />
		<param name="joint_states_freq" value="25" />
		<param name="max_speed_123" value="$(arg max_speed_123)"/>
		<param name="max_speed_456" value="$(arg max_speed_456)"/>
		<param name="use_angle_poses" value="$(arg use_angle_poses)"/>
	</node>

</launch>
